core/src/service/file_service.rs

```diff
@@ -154,6 +155,9 @@ impl<Client: Requester> RequestContext<'_, '_, Client> {
         if tree.calculate_deleted(id)? {
             return Err(CoreError::FileNonexistent);
         }
+        if tree.access_mode(Owner(account.public_key()), id)? < Some(UserAccessMode::Read) {
+            return Err(CoreError::FileNonexistent);
+        }
```

Comment:
don't finalize files you don't have access to e.g. files in deleted pending shares